common/http.hpp

```diff
@@ -261,11 +277,13 @@ class IncomingRequest :
 	void Cancel() override;
 
 private:
-	IncomingRequest(weak_ptr<Stream> stream) :
-		stream_(stream) {
+	IncomingRequest(Stream &stream, shared_ptr<bool> cancelled) :
+		stream_(stream),
+		cancelled_(cancelled) {
 	}
 
-	weak_ptr<Stream> stream_;
+	Stream &stream_;
+	shared_ptr<bool> cancelled_;
```

Comment:
Looking at the resulting code, I think it would make more sense to call the flag `ongoing_` or `active_` and use the opposite logic: initialize to false when creating the object, set to true when operation starts, set to false again on cancel or termination. At this point it seems like we used `cancelled_` "for historical reasons".

It is an observation, I am not saying that we should change it.

Reply:
Yes, I get your point.

But I agree I don't think we should change it now, since the `cancelled_` method is used in a lot of other places too, so at least we're being consistent. Changing it everywhere would be a lot of work, since it's not only a rename, but a logic inversion.